homework3/task4/is_armstrong.py

```diff
@@ -0,0 +1,21 @@
+"""
+Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
+https://en.wikipedia.org/wiki/Narcissistic_number
+Examples:
+- 9 is an Armstrong number, 9 = 9^1 = 9
+- 10 is not: 10 != 1^2 + 0^2 = 1
+- 153 is : 153 = 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153
+Write a function that detects if a number is Armstrong number in functionaly style:
+ - use map or other utilities from functools library,
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+### Example function signature and call
+"""
+
+
+def is_armstrong(number: int) -> bool:
+    digits = list(str(number))
+    if sum([int(i) ** len(digits) for i in digits]) == number:
```

Comment:
Here each time the loop would iterate, `len()` function would be called. How can we simplify it?

Reply:
The same for `int()` call

Reply:
Also, can we somehow use anonymous function here?